dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIKernelBuilderExtensions.cs

```diff
@@ -20,6 +23,149 @@ namespace Microsoft.SemanticKernel;
+
+        foreach (AzureDeploymentInfo d in deployments.Values)
+        {
+            AzureModelInfo model = models[d.Model];
+            string serviceId = $"{serviceIdPrefix ?? string.Empty}{d.Model}";
+
+            if (model.Capabilities.SupportsChatCompletion
+                || model.Id.StartsWith("gpt-", StringComparison.OrdinalIgnoreCase))
+            {
+                builder.WithAzureChatCompletionService(
+                    d.Id,
+                    endpoint,
+                    apiKey,
+                    serviceId: serviceId,
+                    httpClient: httpClient);
+
+                logger?.LogInformation("Registered Azure OpenAI chat completion model: {Model}, serviceId: {ServiceId}", d.Model, serviceId);
+            }
+
+            if (model.Capabilities.SupportsTextCompletion)
+            {
+                builder.WithAzureTextCompletionService(
+                    d.Id,
+                    endpoint,
+                    apiKey,
+                    serviceId: serviceId,
+                    httpClient: httpClient);
+
+                logger?.LogInformation("Registered Azure OpenAI text completion model: {Model}, serviceId: {ServiceId}", d.Model, serviceId);
+            }
+
+            if (model.Capabilities.SupportsEmbeddings)
+            {
+                builder.WithAzureTextEmbeddingGenerationService(
+                    d.Id,
+                    endpoint,
+                    apiKey,
+                    serviceId: serviceId,
+                    httpClient: httpClient);
+
+                logger?.LogInformation("Registered Azure OpenAI embedding generation model: {Model}, serviceId: {ServiceId}", d.Model, serviceId);
+            }
+        }
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds all OpenAI model deployments to the list.
+    /// See https://learn.microsoft.com/azure/cognitive-services/openai for service details.
+    /// </summary>
+    /// <param name="builder">The <see cref="KernelBuilder"/> instance</param>
+    /// <param name="apiKey">OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="organization">OpenAI organization Id</param>
+    /// <param name="serviceIdPrefix">Optional srting to prepend onto the model name when registering the service. Used to disambiguate, when multiple connectors
+    /// may have the same model names. Example: specifying prefix "openai:" will result in model registrations of the form "openai:gpt-3.5-turbo".</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="logger">Application logger</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Self instance</returns>
+    public static KernelBuilder WithOpenAI(this KernelBuilder builder,
```

Comment:
Might be worth supporting an optional string to identify a model as the default.